File: src/main/java/org/funivan/lologa/algo/ga/genome/play/PlayGeneration.java
package org.funivan.lologa.algo.ga.genome.play;

import org.cactoos.iterable.IterableOf;
import org.cactoos.iterable.Joined;
import org.funivan.lologa.algo.ga.genome.GenomeInterface;
import org.funivan.lologa.algo.ga.player.Player;
import org.funivan.lologa.algo.ga.player.PlayerInterface;
import org.funivan.lologa.board.BoardInterface;

public class PlayGeneration implements PlayGenerationInterface {
    private final BoardInterface board;

    public PlayGeneration(BoardInterface board) {
        this.board = board;
    }

    @Override
    public Iterable<PlayerInterface> play(Iterable<GenomeInterface> genomes) {
        Iterable<PlayerInterface> generation = new IterableOf<>();
        for (GenomeInterface genome : genomes) {
            final PlayerInterface max = new Player(genome, this.board);
            generation = new Joined<>(generation, new IterableOf<>(max));
        }
        return generation;
    }

}

File: src/main/java/org/funivan/lologa/algo/ga/genome/play/PlayGeneration.kt
package org.funivan.lologa.algo.ga.genome.play

import org.cactoos.iterable.IterableOf
import org.cactoos.iterable.Joined
import org.funivan.lologa.algo.ga.genome.GenomeInterface
import org.funivan.lologa.algo.ga.player.Player
import org.funivan.lologa.algo.ga.player.PlayerInterface
import org.funivan.lologa.board.BoardInterface

class PlayGeneration(private val board: BoardInterface) : PlayGenerationInterface {

    override fun play(genomes: Iterable<GenomeInterface>): Iterable<PlayerInterface> {
        var generation: Iterable<PlayerInterface> = IterableOf()
        for (genome in genomes) {
            val max = Player(genome, this.board)
            generation = Joined(generation, IterableOf<PlayerInterface>(max))
        }
        return generation
    }

}
